c/CMakeLists.txt

```diff
@@ -464,7 +464,11 @@ set(qpid-proton-noncore-src
   ${qpid-proton-include-extra}
 )
 
-add_library (qpid-proton SHARED $<TARGET_OBJECTS:qpid-proton-core-objects> $<TARGET_OBJECTS:qpid-proton-platform-io-objects> $<TARGET_OBJECTS:qpid-proton-proactor-objects> ${qpid-proton-noncore-src})
+add_library (qpid-proton SHARED
+  $<TARGET_OBJECTS:qpid-proton-core-objects>
+  $<TARGET_OBJECTS:qpid-proton-platform-io-objects>
+  $<$<BOOL:${HAS_PROACTOR}>:$<TARGET_OBJECTS:qpid-proton-proactor-objects>>
```

Comment:
I suggest $<$<TARGET_EXISTS:qpid-proton-proactor-objects>:$<TARGET_OBJECTS:qpid-proton-proactor-objects>>
which would be more future proof.

Reply:
I considered that, but then I thought that `HAS_PROACTOR` is more directly derived from what options the user specified, whereas `$<TARGET_EXISTS` is a check for whatever the build ended up doing. So I felt that HAS_PROACTOR is safer.

I'll change it to TARGET_EXISTS.